### PR TITLE
Fix `minimized-btfs.tar.xz` existence check

### PR DESCRIPTION
**Pull request by usamasaqib**

### CODE REVIEW

File: tasks/kmt.py [hunk 1]
```diff
@@ -653,12 +653,12 @@ def prepare(
...
         d.run_cmd(
             ctx,
-            f"! [ -f {btf_dir}/minimized-btfs.tar.xz \
+            f"! [ -f {btf_dir}/minimized-btfs.tar.xz ] \
                 && [ -d /opt/btf ] \
                 && cd /opt/btf/ \
                 && tar cJf minimized-btfs.tar.xz * \
                 && mkdir -p {btf_dir} \
-                && mv /opt/btf/minimized-btfs.tar.xz {btf_dir}/",
+                && mv /opt/btf/minimized-btfs.tar.xz {btf_dir}/ || [ -f /sys/kernel/btf/vmlinux ] ",
```
[the remark this answers is not in the document]
Comment (usamasaqib) [reply tasks/kmt.py:661]: Actually no, I made a mistake. Then intention is to only setup minimized btf if the kernel does not expose /sys/kernel/btf/vmlinux

Fixing this.

File: tasks/kmt.py [hunk 2]
```diff
@@ -651,15 +651,16 @@ def prepare(
...
-        d.run_cmd(
-            ctx,
-            f"! [ -f {btf_dir}/minimized-btfs.tar.xz ] \
-                && [ -d /opt/btf ] \
-                && cd /opt/btf/ \
-                && tar cJf minimized-btfs.tar.xz * \
-                && mkdir -p {btf_dir} \
-                && mv /opt/btf/minimized-btfs.tar.xz {btf_dir}/ || [ -f /sys/kernel/btf/vmlinux ] ",
-        )
+        if not os.path.exists("/sys/kernel/btf/vmlinux"):
```
Comment (gjulianm) [inline tasks/kmt.py:654]: Watch out because this will be run in the local machine, not the remote domain.

Comment (usamasaqib) [reply tasks/kmt.py]: Fixed. Thanks!

**Review by gjulianm — APPROVED** [review]

File: tasks/kmt.py [hunk 3]
```diff
@@ -651,15 +651,17 @@ def prepare(
...
-                && cd /opt/btf/ \
-                && tar cJf minimized-btfs.tar.xz * \
-                && mkdir -p {btf_dir} \
-                && mv /opt/btf/minimized-btfs.tar.xz {btf_dir}/",
-        )
+        res = d.run_cmd("[ -f /sys/kernel/btf/vmlinux ]", warn=True)
+        if res is None or not res.ok:
+            d.run_cmd(
+                ctx,
+                f"! [ -f {btf_dir}/minimized-btfs.tar.xz ] \
```
Comment (brycekahle) [inline tasks/kmt.py:658]: Could you add `[ -f /sys/kernel/btf/vmlinux ] || ` before this instead?

Comment (usamasaqib) [reply tasks/kmt.py:658]: That would work

**Pull request merged**